### ISSUE TITLE
no recovery notification

### ISSUE THREAD

**Issue by jazzl0ver** [NagiosEnterprises/nagioscore#892]
Hi,

When events happen in the following sequence, there's no service recovery notification sent:

![image](https://user-images.githubusercontent.com/2444175/209680048-09205398-6cde-4790-9f8a-3c3a40641e4e.png)

```
# rpm -q nagios
nagios-4.4.6-4.el7.x86_64

define host{
        host_name                               isp
        use                             generic-host
        alias                           isp
        address                         isp.domain.com
        _HOST_ID                                92
        check_command                   check_centreon_ping!3!200,20%!400,50%
        max_check_attempts                      3
        check_interval                  5
        retry_interval                  1
        check_period                    24x7
        contacts                                <skipped>
        notification_period             24x7
        notification_options            d,u,r,f
        notifications_enabled           1
        stalking_options                        o
}

define command{
        command_name                    check_centreon_ping
        command_line                    $USER1$/check_icmp -H $HOSTADDRESS$ -n $ARG1$ -w $ARG2$ -c $ARG3$
}

define service{
        name                            generic-service
        service_description             generic-service
        is_volatile                     0
        max_check_attempts                      3
        normal_check_interval           5
        retry_check_interval            1
        active_checks_enabled           1
        passive_checks_enabled          0
        check_period                    24x7
        notification_period             24x7
        notification_interval           0
        notification_options            w,u,c,r
        notifications_enabled           1
        contact_groups                  Sysadm-email_only
        register                                0
}

define service{
        host_name                               isp
        service_description             panel alive
        _SERVICE_ID                     461
        _CRITICALITY_LEVEL
        _CRITICALITY_ID
        use                             generic-service
        check_command                   check_nrpe!5666!check_alive_panel
        normal_check_interval           1
        notification_period             24x7
        notification_options            w,u,c,r,f
        contacts                                <skipped>
        stalking_options                        o
}

define command{
        command_name                    check_nrpe
        command_line                    $USER1$/check_nrpe -2 -H $HOSTADDRESS$ -t 30 -p $ARG1$ -c $ARG2$
}
```

**Comment by sawolf** [comment]
Hi @jazzl0ver, thanks for reaching out.

This looks to me like a duplicate of #759, which was fixed in #880/#837 and released as part of Nagios Core 4.4.9. It looks like you're using version 4.4.6 - can you update to the latest version and confirm whether your issue is fixed?

**Comment by jazzl0ver** [comment]
Works great in 4.4.9! Thank you very much!
